include/swift/Reflection/ReflectionContext.h

```diff
@@ -378,30 +378,37 @@ class ReflectionContext
     return readPECOFFSections(ImageStart);
   }
 
-  template <typename T> bool readELFSections(RemoteAddress ImageStart) {
+  template <typename T>
+  bool readELFSections(RemoteAddress ImageStart, const void *FileBuffer,
+                       uint64_t FileSize) {
     auto Buf =
         this->getReader().readBytes(ImageStart, sizeof(typename T::Header));
 
     auto Hdr = reinterpret_cast<const typename T::Header *>(Buf.get());
     assert(Hdr->getFileClass() == T::ELFClass && "invalid ELF file class");
 
-    // From the header, grab informations about the section header table.
-    auto SectionHdrAddress = ImageStart.getAddressData() + Hdr->e_shoff;
+    auto FileBufferStartAddress = (uint64_t) FileBuffer;
```

Comment:
`uint64_t FileBufferStartAddress = (uint64_t)FileBuffer;` ?

Reply:
I think I would prefer explicit types for most of these variables here. (Personal preference)

Reply:
> uint64_t FileBufferStartAddress = (uint64_t)FileBuffer; 

I cast it to a number just so I can avoid doing it later in 3 different places.

Reply:
This was about the `auto`. I would use explicit type names for most of the local variables in the function, unless the type is abundantly clear from the context (i.e., the cast on line 385 and in a for loop), or if the type name is so long that spelling it out would make the code less readable.

Reply:
Oh! Ok!

Reply:
I used explicit types where I thought it made sense, and where the type didn't depend on the template.